code/modules/mob/living/living.dm

```diff
@@ -715,38 +715,42 @@ Thanks.
 							pulling.Move(T, get_dir(pulling, T))
 							if(M && secondarypull)
 								M.start_pulling(secondarypull)
+							
 							/* Drag damage is here!*/
 							var/mob/living/carbon/human/HM = M
-							var/list/damaged_organs = HM.drag_damage()
-							if (damaged_organs)
-								if(!HM.isincrit() && damaged_organs.len)
-									if(prob(HM.getBruteLoss() / 5)) //Chance to damage
+							var/list/damaged_organs = HM.get_broken_organs()
+							var/list/bleeding_organs = HM.get_bleeding_organs()
+							
+							if (damaged_organs.len)
+								if(!HM.isincrit())
+									if(prob(HM.getBruteLoss() / 5)) //Chance for damage based on current damage
 										for(var/datum/organ/external/damagedorgan in damaged_organs)
-											if((damagedorgan.brute_dam) < damagedorgan.max_damage)
+											if((damagedorgan.brute_dam) < damagedorgan.max_damage) //To prevent organs from accruing thousands of damage
 												HM.apply_damage(2, BRUTE, damagedorgan)
 												HM.visible_message("<span class='warning'>The wounds on \the [HM]'s [damagedorgan.display_name] worsen from being dragged!</span>")
 												HM.UpdateDamageIcon()
-									if(prob(HM.getBruteLoss() / 8)) //Chance to bleed
-										blood_splatter(HM.loc,HM)
-										var/blood_volume = round(HM:vessel.get_reagent_amount("blood"))
-										if(blood_volume > 0)
-											HM:vessel.remove_reagent("blood",4)
-											HM.visible_message("<span class='warning'>\The [HM] loses some blood from being dragged!</span>")
-
-								if(HM.isincrit() && damaged_organs.len) //Crit damage boost
+								else
 									if(prob(15))
 										for(var/datum/organ/external/damagedorgan in damaged_organs)
 											if((damagedorgan.brute_dam) < damagedorgan.max_damage)
 												HM.apply_damage(4, BRUTE, damagedorgan)
 												HM.visible_message("<span class='warning'>The wounds on \the [HM]'s [damagedorgan.display_name] worsen terribly from being dragged!</span>")
 												add_logs(src, HM, "caused drag damage to", admin = (M.ckey))
 												HM.UpdateDamageIcon()
-									if(prob(8))
-										if(isturf(HM.loc))
-											blood_splatter(HM.loc,HM,1)
-											var/blood_volume = round(HM:vessel.get_reagent_amount("blood"))
-											if(blood_volume > 0)
-												HM:vessel.remove_reagent("blood",8)
+							
+							if (bleeding_organs.len)
+								var/blood_volume = round(HM:vessel.get_reagent_amount("blood"))
+								if(blood_volume > 0 || (HM.species.anatomy_flags & NO_BLOOD))
```

Comment:
Yeah. Because people with NO_BLOOD and people with a blood that isn't 0 aren't always the same people.